pkg/controller/configmapcainjector/controller/sa_token_cabundle_controller.go

```diff
@@ -0,0 +1,77 @@
+package controller
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	informers "k8s.io/client-go/informers/core/v1"
+	kcoreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	listers "k8s.io/client-go/listers/core/v1"
+
+	"bytes"
+
+	"github.com/openshift/service-serving-cert-signer/pkg/boilerplate/controller"
+	"github.com/openshift/service-serving-cert-signer/pkg/controller/api"
+)
+
+// saTokenCABundleInjectionController is responsible for injecting a CA bundle into service account token secrets.
+type saTokenCABundleInjectionController struct {
+	secretsClient kcoreclient.SecretsGetter
+	secretsLister listers.SecretLister
+
+	ca []byte
+}
+
+func secretIsSAToken(obj v1.Object) bool {
+	secret, ok := obj.(*corev1.Secret)
```

Comment:
`return obj.(*corev1.Secret).Type == corev1.SecretTypeServiceAccountToken`

and use directly as `AddFunc`, and call it in `UpdateFunc`